controllers/common.go

```diff
@@ -661,3 +661,58 @@ func (r *VolumeSnapshotRestoreReconciler) setVSRQueue(vsr *volsnapmoverv1alpha1.
 
 	return true, nil
 }
+
+func (r *VolumeSnapshotBackupReconciler) GetRepSrcPodSecurityContext(namespace string, sourcePVCName string) (*corev1.PodSecurityContext, error) {
+
+	podSC := corev1.PodSecurityContext{}
+	podList := corev1.PodList{}
+	if err := r.List(r.Context, &podList, &client.ListOptions{Namespace: namespace}); err != nil {
+		return nil, err
+	}
+
+	for _, pod := range podList.Items {
+		if pod.Spec.Volumes != nil {
+			po := getPVCPod(&pod, sourcePVCName)
+
+			if po != nil && po.Spec.SecurityContext != nil {
+				podSC = *po.Spec.SecurityContext
+				return &podSC, nil
+			}
+		}
+	}
+
+	return nil, nil
```

Comment:
Would this be more performant? make list return a smaller list.
```suggestion
	if err := r.List(r.Context, &podList, &client.ListOptions{
		Namespace: namespace,
		FieldSelector: 
			fields.SelectorFromSet(fields.Set{
				"spec.volumes.persistentVolumeClaim.claimName": sourcePVCName,
			}),
	}); err != nil {
		return nil, err
	}

	for _, pod := range podList.Items {
		if pod.Spec.SecurityContext != nil {
			podSC = *pod.Spec.SecurityContext
			return &podSC, nil
		}
	}

	return nil, nil
```

Reply:
nodeName would work tho as a fieldselector. but since we're already scoped to namespace it wouldn't be absolutely necessary.